cli/internal/lockfile/pnpm_lockfile.go

```diff
@@ -180,20 +181,52 @@ func (p *PnpmLockfile) Subgraph(packages []string) (Lockfile, error) {
 		}
 	}
 
+	importers, err := pruneImporters(&p.Importers, workspacePackages)
+
+	if err != nil {
+		return nil, err
+	}
+
 	lockfile := PnpmLockfile{
 		Version:                   p.Version,
-		Importers:                 p.Importers,
+		Importers:                 importers,
 		Packages:                  lockfilePackages,
 		NeverBuiltDependencies:    p.NeverBuiltDependencies,
 		OnlyBuiltDependencies:     p.OnlyBuiltDependencies,
 		Overrides:                 p.Overrides,
 		PackageExtensionsChecksum: p.PackageExtensionsChecksum,
-		PatchedDependencies:       p.PatchedDependencies,
+		// TODO only the applicable patches should be copied to the subgraph
+		// before we can implement this we need to be able to prune the pnpm section
+		// of package.json otherwise installation will fail
+		PatchedDependencies: p.PatchedDependencies,
 	}
 
 	return &lockfile, nil
 }
 
+// Prune imports to only those have all of their dependencies in the packages list
+func pruneImporters(importers *map[string]ProjectSnapshot, workspacePackages []turbopath.AnchoredSystemPath) (map[string]ProjectSnapshot, error) {
+	prunedImporters := map[string]ProjectSnapshot{}
+
+	// Copy over root level importer
+	if root, ok := (*importers)["."]; ok {
+		prunedImporters["."] = root
+	}
+
+	for _, workspacePath := range workspacePackages {
+		workspace := workspacePath.ToUnixPath().ToString()
```

Comment:
pnpm encodes workspace paths using `/` as a separator regardless of the separator the OS uses